### PR TITLE
[CCR] Validate remote cluster license as part of put auto follow pattern api call

### PR DESCRIPTION
**Pull request by martijnvg**
in addition of validation that when auto follow coordinator starts auto
following indices in the leader cluster.

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutAutoFollowPatternAction.java [hunk 1]
```diff
@@ -75,41 +76,40 @@ protected void masterOperation(PutAutoFollowPatternAction.Request request,
             return;
         }
         final Client leaderClient = client.getRemoteClusterClient(request.getRemoteCluster());
```
Comment (jasontedor) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutAutoFollowPatternAction.java]: leader -> remote, in a few places.

File: x-pack/plugin/ccr/qa/multi-cluster-with-non-compliant-license/src/test/java/org/elasticsearch/xpack/ccr/CcrMultiClusterLicenseIT.java [hunk 2]
```diff
@@ -35,32 +30,14 @@ public void testAutoFollow() throws Exception {
         if ("follow".equals(targetCluster)) {
             final Request request = new Request("PUT", "/_ccr/auto_follow/test_pattern");
             request.setJsonEntity("{\"leader_index_patterns\":[\"*\"], \"remote_cluster\": \"leader_cluster\"}");
-            client().performRequest(request);
-
-            // parse the logs and ensure that the auto-coordinator skipped coordination on the leader cluster
-            assertBusy(() -> {
```
Comment (jasontedor) [inline x-pack/plugin/ccr/qa/multi-cluster-with-non-compliant-license/src/test/java/org/elasticsearch/xpack/ccr/CcrMultiClusterLicenseIT.java]: I think we still might want a test for this, for the case when the license expires upstream.

**Review by jasontedor — COMMENTED** [review]
I left one nit. I’m doubting removing the existing test. Thanks for picking this up.

File: x-pack/plugin/ccr/qa/multi-cluster-with-non-compliant-license/src/test/java/org/elasticsearch/xpack/ccr/CcrMultiClusterLicenseIT.java (hunk 2 above)
Comment (martijnvg) [reply x-pack/plugin/ccr/qa/multi-cluster-with-non-compliant-license/src/test/java/org/elasticsearch/xpack/ccr/CcrMultiClusterLicenseIT.java]: Yes, that makes sense. I just wonder how we can test this. We need to start a cluster with a platinum license and then downgrade to a basic license? But that would require another qa module, right?

Comment (martijnvg) [reply x-pack/plugin/ccr/qa/multi-cluster-with-non-compliant-license/src/test/java/org/elasticsearch/xpack/ccr/CcrMultiClusterLicenseIT.java]: Now I think more about it, a qa module that downgrades the remote cluster, is also useful to test what happens with an index being followed.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**